File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def solution(numbers, target):
    def dfs(index, total):
        if index == len(numbers):
            if total == target:
                return 1
            else:
                return 0
        return dfs(index+1, total + numbers[index]) + dfs(index+1, total- numbers[index])
    
    return dfs(0,0)

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
import java.util.*;

class Solution {
    static int cnt = 0;
    static int N = 0;
    public int solution(int[] numbers, int target) {
        N = numbers.length;
        binary(numbers, 0, target, 0);
        return cnt;
    }
    
    void binary(int[] numbers, int idx, int target, int out){
        if(idx == N){
            if(out == target) cnt ++;
            return;
        }
        
        binary(numbers, idx+1, target, out+ numbers[idx]);
        binary(numbers, idx+1, target, out- numbers[idx]);
        
        return;
    }
}